tests/rewriter_test.py

```diff
@@ -194,3 +194,79 @@ def transformation(module: ModuleOp, rewriter: Rewriter) -> None:
         rewriter.inline_block_after(if_block, constant_op)
 
     rewrite_and_compare(prog, expected, transformation)
+
+
+def test_insert_block():
+    """Test the insertion of a block in a region."""
+    prog = \
+    """module() {
+  %0 : !i1 = arith.constant() ["value" = 1 : !i1]
+}"""
+
+    expected = \
+"""module() {
+^0: ^1: 
+  %0 : !i1 = arith.constant() ["value" = 1 : !i1]
+}"""
+
+    def transformation(module: ModuleOp, rewriter: Rewriter) -> None:
+        module.regions[0].insert_block(Block(), 0)
+
+    rewrite_and_compare(prog, expected, transformation)
+
+
+def test_insert_block2():
+    """Test the insertion of a block in a region."""
+    prog = \
+    """module() {
+  %0 : !i1 = arith.constant() ["value" = 1 : !i1]
+}"""
+
+    expected = \
+"""module() {
+^0: 
+  %0 : !i1 = arith.constant() ["value" = 1 : !i1]
+^1: }"""
+
+    def transformation(module: ModuleOp, rewriter: Rewriter) -> None:
+        module.regions[0].insert_block(Block(), 1)
+
+    rewrite_and_compare(prog, expected, transformation)
+
+
+def test_insert_block_before():
+    """Test the insertion of a block before another block."""
+    prog = \
+    """module() {
+  %0 : !i1 = arith.constant() ["value" = 1 : !i1]
+}"""
+
+    expected = \
+"""module() {
+^0: ^1: 
+  %0 : !i1 = arith.constant() ["value" = 1 : !i1]
+}"""
+
+    def transformation(module: ModuleOp, rewriter: Rewriter) -> None:
+        rewriter.insert_block_before(Block(), module.regions[0].blocks[0])
+
+    rewrite_and_compare(prog, expected, transformation)
+
+
```

Comment:
Maybe add a test that has more than one initial block for both insert before and after.